Interface/CustomDecoder.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace Kaitai
+{
+    /// <summary>
+    ///     A custom decoder interface. Implementing classes can be called from
+    ///     inside a .ksy file using `process: XXX` syntax.
+    /// </summary>
```

Comment:
Take a look at my new architecture for python runtime. https://gitlab.com/KOLANICH/Cache.py/tree/master/Cache/kaitai/compress . It may worth to mirror it here.

Reply:
Let's improve this in follow-ups.